server/src/main/java/org/apache/cassandra/sidecar/routes/AbstractHandler.java

```diff
@@ -290,22 +292,31 @@ private Name tableName(RoutingContext context, boolean required)
      * Given a combined host address like 127.0.0.1:9042 or [2001:db8:0:0:0:ff00:42:8329]:9042, this method
      * removes port information and returns 127.0.0.1 or 2001:db8:0:0:0:ff00:42:8329.
      *
-     * @param address ip address
+     * @param request http server request
      * @return host address without port information
+     * @throws NoSuchCassandraInstanceException thrown when input address is null
      */
-    public static String extractHostAddressWithoutPort(String address)
```

Comment:
The original input `address` is nullable. But it is not handled. So the existing behavior is throwing NPE, leading to 503 internal server error. It is unexpected and no client side logic relies on this behavior. 

In this patch, `null` value is handled, and server returns 404 not found.

Regardless of the handling change, the `handlers` are only going to see non-null `host` value, in both cases. The fact is clarified with adding the `@NotNull` annotation.